htdocs/core/class/commonobject.class.php

```diff
@@ -3898,7 +3906,7 @@ public function update_price($exclspec = 0, $roundingadjust = 'none', $nodatabas
 						}
 						$obj->total_tva = $tmpcal[1];
 						$obj->total_ttc = $tmpcal[2];
-					} elseif ($diff_when_using_price_ht && $roundingadjust == '0') {
+					} elseif ($diff_when_using_price_ht) {
```

Comment:
Why removing the test ?
If we call the method with 'none', it means we don't want to make data cleaning. 
By removing the test, we will execute this even when using 'none'

Reply:
@eldy
"roundingajust" was not the good variable to use here and I replaced this condition with `$forcedroundingmode == '0'` but it was already checked at line 3889 : `if (empty($reshook) && $forcedroundingmode == '0') {`
Otherwise when you got "auto" in roudingadjust parameter, total amounts weren't updated even if a difference was detected.
So if we have `roundingajust = 'none'` as parameter, it will never goes to this part of code because forcedroundingmode was set to 'none" at line 3816 : `$forcedroundingmode = $roundingadjust"`